src/errors.ts

```diff
@@ -0,0 +1,112 @@
+import type { IncomingHttpHeaders } from "node:http";
+
+export type TimeoutEvent =
+  | "request"
+  | "response"
+  | "read"
+  | "socket"
+  | "lookup"
+  | "connect";
+
+export class StdError extends Error {
+  code?: string;
+  host?: string;
+  hostname?: string;
+  method?: string;
+  path?: string;
+  protocol?: string;
+  url?: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any -- This is a generic error
+  response?: any;
+}
+
+export class RequestError extends StdError {
+  name: "RequestError" = "RequestError";
+}
+
+export class ReadError extends StdError {
+  name: "ReadError" = "ReadError";
+}
+
+export class ParseError extends StdError {
+  name: "ParseError" = "ParseError";
+  statusCode: number;
+  statusMessage: string;
+
+  constructor(
+    message: string,
+    statusCode: number,
+    statusMessage: string
+  ) {
+    super(message);
+    this.statusCode = statusCode;
+    this.statusMessage = statusMessage;
+  }
+}
+
+export class HTTPError extends StdError {
+  name: "HTTPError" = "HTTPError";
+  statusCode: number;
+  statusMessage: string;
+  headers: IncomingHttpHeaders;
+  body: Buffer | string | object;
```

Comment:
I'm not against this, but this should be listed in the breaking changes.

Currently it is on `.response.body`, it will become `.body`

We could rework this to be more compatible by having:

```
response: { headers: IncomingHttpHeaders; body: Buffer | string | object; }
```

Reply:
Also what about empty body responses (eg 204).

I feel like a `null` body for a 204 is more correct